File: .claude/hooks/user-prompt-name-agent.mjs
#!/usr/bin/env node
// @hookstack user-prompt-llm-agent-name
// Attribue un nom à l'agent pour la session courante (UserPromptSubmit)
import { readFileSync, writeFileSync, mkdirSync, existsSync } from 'node:fs';
import { join } from 'node:path';
import { homedir } from 'node:os';
import { fileURLToPath } from 'node:url';

const NAMES = ['Phoenix', 'Sage', 'Nova', 'Echo', 'Atlas', 'Cipher', 'Nexus', 'Oracle', 'Aurora', 'Vortex'];

export function run(
  input,
  {
    exists = existsSync,
    readFile = readFileSync,
    writeFile = writeFileSync,
    mkdir = mkdirSync,
    home = homedir(),
    pickName = () => NAMES[Math.floor(Math.random() * NAMES.length)],
  } = {},
) {
  const sessionId = input.session_id ?? 'unknown';
  const dir = join(home, '.claude', 'data', 'sessions');
  mkdir(dir, { recursive: true });

  const file = join(dir, `${sessionId}.json`);
  let data = { session_id: sessionId };
  if (exists(file)) {
    try { data = JSON.parse(readFile(file, 'utf8')); } catch {}
  }

  if (data.agent_name) return null;

  data.agent_name = pickName();
  writeFile(file, JSON.stringify(data, null, 2));
  return `Tu t'appelles **${data.agent_name}** pour cette session.\n`;
}

/* v8 ignore next 5 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  const result = run(input);
  if (result) process.stdout.write(result);
}

File: .claude/hooks/user-prompt-name-agent.py
#!/usr/bin/env python3
# @hookstack user-prompt-llm-agent-name
"""Assigns a name to the agent for the current session (UserPromptSubmit)."""
import json
import os
import random
import sys

NAMES = [
    "Phoenix",
    "Sage",
    "Nova",
    "Echo",
    "Atlas",
    "Cipher",
    "Nexus",
    "Oracle",
    "Aurora",
    "Vortex",
]


def run(
    input_data=None,
    *,
    exists=None,
    read_file=None,
    write_file=None,
    mkdir=None,
    home=None,
    pick_name=None,
):
    if exists is None:
        exists = os.path.exists
    if read_file is None:
        read_file = lambda p: open(p, "r", encoding="utf8").read()
    if write_file is None:
        write_file = lambda p, content: open(p, "w", encoding="utf8").write(content)
    if mkdir is None:
        mkdir = os.makedirs
    if home is None:
        home = os.path.expanduser("~")
    if pick_name is None:
        pick_name = lambda: random.choice(NAMES)

    session_id = (input_data or {}).get("session_id") or "unknown"
    dir_path = os.path.join(home, ".claude", "data", "sessions")
    try:
        mkdir(dir_path, exist_ok=True)
    except Exception:
        pass

    file_path = os.path.join(dir_path, f"{session_id}.json")
    data = {"session_id": session_id}
    if exists(file_path):
        try:
            data = json.loads(read_file(file_path))
        except Exception:
            pass

    if data.get("agent_name"):
        return None

    data["agent_name"] = pick_name()
    write_file(file_path, json.dumps(data, indent=2))
    return f"Tu t'appelles **{data['agent_name']}** pour cette session.\n"


if __name__ == "__main__":
    data = json.load(sys.stdin)
    result = run(data)
    if result:
        sys.stdout.write(result)
